src/sqlfluff/testing/rules.py

```diff
@@ -120,6 +120,42 @@ def assert_rule_raises_violations_in_file(rule, fpath, violations, fluff_config)
     assert set(lnt.check_tuples()) == {(rule, v[0], v[1]) for v in violations}
 
 
+def prep_violations(rule, violations):
+    """Default to test rule if code is omitted."""
+    for v in violations:
+        if "code" not in v:
+            v["code"] = rule
+    return violations
+
+
+def assert_violations_before_fix(test_case, violations_before_fix):
+    """Assert that the given violations are found in the given sql."""
+    violation_info = [e.get_info_dict() for e in violations_before_fix]
+    try:
+        assert violation_info == prep_violations(test_case.rule, test_case.violations)
+    except AssertionError:
+        print("Actual violations:\n" + yaml.dump(violation_info))
+        raise
```

Comment:
Shouldn't this be covered by one of your tests below?

Reply:
Could you elaborate what you mean?

Reply:
So the test coverage has dropped below 100% so we can't merge this. This line (and a similar exception below) are the reason.

So we either need to add a test case, or add a `  # pragma: no cover` to this exception (which isn't unusual for exceptions to be honest as they probably shouldn't really be reached except in, exceptional circumstances).

But curious why this area of code is not reached anyway since you have a test case for extra violations not being reached?

Reply:
Ah, thanks, so it's about the coverage of "test framework".

I'll simplify the code as said here, that should fix the coverage problem: https://github.com/sqlfluff/sqlfluff/pull/2718#discussion_r815450507

Reply:
Btw, to test coverage locally, is this the minimal scope that I need to run:

```
tox -e generate-fixture-yml,cov-init,py38,cov-report
```
?

Just wondering because for me the full py38 test suite takes quite long to run locally. I noticed that my IDE respects the `# pragma: no cover` comments so that is handy though, and let's me check individual cases quickly. But I'm looking for a one-liner to perform the "100% coverage" check locally. So that I can know that it will also pass on CI.

Reply:
That should be enough but I've noticed some lines aren't covered by that anymore (and haven't been for a while!) 😞 Been meaning to find some time to look at it but running locally is slow enough already so end up using CI for this check myself to be honest.

Reply:
Ouch, ok. So even more tox targets would be needed to reach the 100%.

Reply:
Yeah think there is a windows specific check now for file paths, and also some dbt tests. Hency why I haven't priortised looking into this 🤨

Reply:
Looks like the "pragma: no cover" worked. The coverage report shows 100%: https://app.codecov.io/gh/sqlfluff/sqlfluff/compare/2718?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr%20comments&utm_term=sqlfluff